core/services/ardupilot_manager/ArduPilotManager.py

```diff
@@ -258,3 +292,53 @@ def remove_endpoints(self, endpoints_to_remove: Set[Endpoint]) -> None:
                 raise
 
         self._update_endpoints()
+
+    def get_available_firmwares(self, vehicle: Vehicle) -> List[str]:
+        try:
+            return self.firmware_download.get_available_versions(vehicle, self.current_platform)
+        except Exception as error:
+            logger.exception(f"Unable to retrieve available firmwares: {error}\n")
+            raise
+
+    def install_custom_firmware(self, firmware_path: pathlib.Path) -> None:
+        try:
+            self.mavlink_manager.stop()
+            self.prune_ardupilot_processes()
+            self.firmware_installer.install_firmware(firmware_path)
+            logger.info(f"Succefully installed firmware for {self.current_platform}.")
+        except Exception as error:
+            logger.exception(f"Unable to install firmware: {error}\n")
+            raise
+        finally:
+            self.restart()
+
+    def install_custom_firmware_version(self, vehicle: Vehicle, version: str) -> None:
+        try:
+            self.mavlink_manager.stop()
+            self.prune_ardupilot_processes()
+            temporary_file = self.firmware_download.download(vehicle, self.current_platform, version)
+            assert temporary_file, "Failed to download firmware file."
+            self.firmware_installer.install_firmware(temporary_file)
+            logger.info(f"Succefully installed firmware version {version} for {self.current_platform}/{vehicle}.")
+        except Exception as error:
+            logger.exception(f"Unable to install firmware: {error}\n")
+            raise
+        finally:
+            self.restart()
+
+    def install_latest_firmware(self, vehicle: Vehicle, update: bool = True) -> None:
```

Comment:
Update is not clear, looking the code it appears to be more a "force", can you explain or add a comment about it ?

Reply:
Yes, it's a force update, although I've changed the behavior for the Installer class and kept this there. I will change a little bit the behavior so it gets better.

Reply:
Done.